Tasks/VsTest/versionfinder.ts

```diff
@@ -7,28 +7,28 @@ import utils = require('./helpers');
 var regedit = require('regedit');
 var xml2js = require('xml2js');
 
-export function locateVSTestConsole(testConfig): Q.Promise<models.ExecutabaleInfo> {
-    let deferred = Q.defer<models.ExecutabaleInfo>();
-    locateTestWindow(testConfig).
-        then( function(exeInfo) {
-            var vstestConsoleInfo = exeInfo;
-            if(exeInfo){
-                vstestConsoleInfo.location = path.join(exeInfo.location, "vstest.console.exe");
-            }            
-            deferred.resolve(vstestConsoleInfo);
-        });
+export async function locateVSTestConsole(testConfig): Promise<string> {
+    let deferred = Q.defer<string>();
+    let vstestExeFolder = await locateTestWindow(testConfig);
+        
+    var vstestExePath = vstestExeFolder;
+    if(vstestExeFolder){
+        vstestExePath = path.join(vstestExeFolder, "vstest.console.exe");
+    }
+
+    deferred.resolve(vstestExePath);
     return deferred.promise;
 }
 
-export function locateTestWindow(testConfig: models.TestConfigurations): Q.Promise<models.ExecutabaleInfo> {
-    let deferred = Q.defer<models.ExecutabaleInfo>();
+export function locateTestWindow(testConfig: models.TestConfigurations): Q.Promise<string> {
```

Comment:
there's fundamental difference between Q.Promise vs Promise. 

Q is external npm lib
Promise is ES6 standard lib and always use Promise and get rid of Q.Promise